sql/src/main/java/io/crate/analyze/AnalyzedTableElements.java

```diff
@@ -158,6 +181,76 @@ public void finalizeAndValidate() {
         validatePrimaryKeys();
     }
 
+    private void validateGeneratedColumns(TableIdent tableIdent,
+                                          AnalysisMetaData analysisMetaData,
+                                          ParameterContext parameterContext) {
+        List<ReferenceInfo> tableReferenceInfos = new ArrayList<>();
+        for (AnalyzedColumnDefinition columnDefinition : columns) {
+            buildReferenceInfo(tableIdent, columnDefinition, tableReferenceInfos);
+        }
+        try {
+            TableInfo tableInfo = analysisMetaData.referenceInfos().getTableInfo(tableIdent);
```

Comment:
would it be possible to pass a nullable TableInfo into validateGeneratedColumns (in the alter table case there is a tableInfo already available somewhere) ?